pkg/machine/e2e/proxy_test.go

```diff
@@ -0,0 +1,62 @@
+package e2e_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("podman machine proxy settings propagation", func() {
+	var (
+		mb      *machineTestBuilder
+		testDir string
+	)
+
+	BeforeEach(func() {
+		testDir, mb = setup()
+	})
+	AfterEach(func() {
+		teardown(originalHomeDir, testDir, mb)
+	})
+
+	It("ssh to running machine and check proxy settings", func() {
+		name := randomString()
+		i := new(initMachine)
+		session, err := mb.setName(name).setCmd(i.withImagePath(mb.imagePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		proxyURL := "http://abcdefghijklmnopqrstuvwxyz-proxy"
+		if err := os.Setenv("HTTP_PROXY", proxyURL); err != nil {
+			Fail("failed to set HTTP_PROXY")
+		}
+		if err := os.Setenv("HTTPS_PROXY", proxyURL); err != nil {
+			Fail("failed to set HTTPS_PROXY")
+		}
+
+		s := new(startMachine)
+		startSession, err := mb.setName(name).setCmd(s).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(startSession).To(Exit(0))
+
+		sshProxy := sshMachine{}
+		sshSession, err := mb.setName(name).setCmd(sshProxy.withSSHCommand([]string{"printenv", "HTTPS_PROXY"})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sshSession).To(Exit(0))
+		Expect(sshSession.outputToString()).To(ContainSubstring(proxyURL))
+
+		sshSession, err = mb.setName(name).setCmd(sshProxy.withSSHCommand([]string{"printenv", "HTTPS_PROXY"})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sshSession).To(Exit(0))
+		Expect(sshSession.outputToString()).To(ContainSubstring(proxyURL))
+
+		if err := os.Unsetenv("HTTP_PROXY"); err != nil {
+			Fail("unable to unset HTTP_PROXY")
+		}
+		if err := os.Unsetenv("HTTPS_PROXY"); err != nil {
+			Fail("unable to unset HTTPS_PROXY")
+		}
```

Comment:
Thanks, please use `defer os.Unsetenv(...)` instead directly after the setenv call.
The thing with ginkgo is that if a match above fails it stops executing this function and returns early, so if this test fails you will leak the env into all following test causing likely more errors.

Reply:
Done
For some unclear for me reason, a simple definition of `defer os.Unsetenv("...")` didn't work and [there were leaks of the env into all following tests](https://cirrus-ci.com/task/4641321554542592?logs=main#L1093)
Now, there is a slightly bulky defer in the test, but it works, it's simple and it's clear.